### PR TITLE
fix(table-of-contents): adjustment to z-index on table of contents styles

### PR DESCRIPTION
**Pull request by m4olivei**
### Related Ticket(s)

Resolves https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/9949 
Jira: https://jsw.ibm.com/browse/ADCMS-2594
Jira: https://jsw.ibm.com/browse/ADCMS-2829
Jira: https://jsw.ibm.com/browse/ADCMS-2488

### Description

Bumps the `z-index` on the Table of contents (TOC) so that the mobile view rises above components like carousel's and accordions that declare a modest `z-index`.

To test, make use of the Web Components Code Sandbox Examples e2e testing environment that was built for this PR:

https://webcomponents-codesandbox.s3-web.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/9957/index.html

Switch to a mobile breakpoint where the TOC becomes sticky on scroll. Verify that the TOC remains above both the accordion (https://jsw.ibm.com/browse/ADCMS-2594) and the carousel (https://jsw.ibm.com/browse/ADCMS-2829).

### Changelog

**Changed**

- Bump `z-index` in the TOC styles.

### CODE REVIEW

**Comment by m4olivei** [comment]
The e2e test failure _appears_ to be unrelated, as it's complaining about the CTA link video component. I've pinged @andy-blum about that. As I also see the issue [in Canary](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/iframe.html?knob-CTA%20style%20(cta-style)=card-link&knob-CTA%20type%20(cta-type)=video&knob-Copy%20(copy):=Lorem%20ipsum%20dolor%20sit%20amet&knob-Content%20link%20href%20(href)=https://www.example.com&knob-Heading%20(heading):=Explore%20AI%20use%20cases%20in%20all%20industries&knob-Custom%20video%20title=Custom%20video%20title&knob-Custom%20video%20description=This%20is%20a%20custom%20video%20description&knob-Video%20ID%20(href)=1_9h94wo6b&knob-Download%20target%20(download)=IBM_Annual_Report_2019.pdf&knob-Download%20link%20href%20(href)=https://www.ibm.com/annualreport/assets/downloads/IBM_Annual_Report_2019.pdf&id=components-cta--default&viewMode=story). The test is looking for the play icon in the component, and while it does appear for a split second, it's quickly removed from the DOM, hence failing [the test](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/tests/e2e-storybook/cypress/integration/cta/card-link.e2e.js#L86).

File: packages/web-components/examples/codesandbox/components/table-of-contents/index.html [hunk 1]
```diff
@@ -25,7 +25,22 @@
...
+
+    dds-carousel {
+      --dds--carousel--overflow-hint-size: 0;
+      padding: 2rem 0;
+    }
   </style>
+  <!-- CDN artifact -->
+  <script
+    type="module"
+    src="https://1.www.s81c.com/common/carbon/carbon-for-aem/tag/canary/accordion.min.js"
```
Comment (proeung) [inline packages/web-components/examples/codesandbox/components/table-of-contents/index.html:42]:
```suggestion
    src="https://1.www.s81c.com/common/carbon/carbon-for-aem/tag/latest/accordion.min.js"
```

Minor/nitpicking - Let's use the `latest` tag for the `<caem-accordion>` component.

**Review by proeung — COMMENTED** [review]
I just have a minor comment, but everything looks good. TOC no longer overlaps on other components within the mobile breakpoints (see screenshot).

https://user-images.githubusercontent.com/1815714/217077633-6d16d9f0-def4-496a-a706-96e3ef8054a6.mov

**Review by proeung — APPROVED** [review]

**Review by andy-blum — APPROVED** [review]
e2e tests are passing now.

**Comment by proeung** [comment]
@carbon-design-system/carbon-for-ibm-com-members Can we get a review so that this PR can be merged down? Thanks!

**Review by annawen1 — APPROVED** [review]
LGTM! Thanks for adding to the codesandbox example as well!

**Comment by proeung** [comment]
So close! @annawen1 Can I get your review on the required Percey builds?

**Comment by m4olivei** [comment]
> Can I get your review on the required Percey builds?

@proeung / @annawen1 For my own edification, why is Percy catching something here? It looks like it's complaining about the Components/Tag link: Default, which wasn't touched here. Is it just a quirk of Percy that this kind of thing sometimes pops up?

**Pull request merged**